src/rpc_server.rs

```diff
@@ -204,150 +204,944 @@ pub trait RPC {
+    /// # tokio_test::block_on(async {
+    ///
+    /// // create a serde/json transport over tcp.
+    /// let transport = tcp::connect("127.0.0.1:9799", Json::default).await.unwrap();
+    ///
+    /// // create an rpc client using the transport.
+    /// let client = RPCClient::new(client::Config::default(), transport).spawn();
+    ///
+    /// // query neptune-core server how to find the cookie file
+    /// let cookie_hint = client.cookie_hint(context::current()).await.unwrap().unwrap();
+    ///
+    /// # })
+    /// ```
     /// this method does not require authentication because local clients must
     /// be able to call this method in order to bootstrap cookie-based
     /// authentication.
+    ///
     async fn cookie_hint() -> RpcResult<rpc_auth::CookieHint>;
 
     /// Return the network this neptune-core instance is running
     ///
-    /// this method does not require authentication because local clients must
-    /// be able to call this method (in fallback mode) in order to bootstrap
-    /// cookie-based authentication
+    /// ```no_run
+    /// use neptune_cash::rpc_server::RPCClient;
+    /// use neptune_cash::rpc_auth;
+    /// use tarpc::tokio_serde::formats::Json;
+    /// use tarpc::serde_transport::tcp;
+    /// use tarpc::client;
+    /// use tarpc::context;
+    ///
+    /// # tokio_test::block_on(async {
+    ///
+    /// // create a serde/json transport over tcp.
+    /// let transport = tcp::connect("127.0.0.1:9799", Json::default).await.unwrap();
+    ///
+    /// // create an rpc client using the transport.
+    /// let client = RPCClient::new(client::Config::default(), transport).spawn();
+    ///
+    /// // query neptune-core server the network it is running on.
+    /// let network = client.network(context::current()).await.unwrap().unwrap();
     ///
-    /// Fallback mode occurs when the cookie_hint() RPC method is disabled or
-    /// if end user specifies location of the cookie file manually.
+    /// # })
+    /// ```
     async fn network() -> RpcResult<Network>;
 
     /// Returns local socket used for incoming peer-connections. Does not show
     /// the public IP address, as the client does not know this.
+    ///
+    /// ```no_run
+    /// use neptune_cash::rpc_server::RPCClient;
+    /// use neptune_cash::rpc_auth;
+    /// use tarpc::tokio_serde::formats::Json;
+    /// use tarpc::serde_transport::tcp;
+    /// use tarpc::client;
+    /// use tarpc::context;
+    ///
+    /// # tokio_test::block_on(async {
+    ///
+    /// // create a serde/json transport over tcp.
+    /// let transport = tcp::connect("127.0.0.1:9799", Json::default).await.unwrap();
+    ///
+    /// // create an rpc client using the transport.
+    /// let client = RPCClient::new(client::Config::default(), transport).spawn();
+    ///
+    /// // load the cookie file from disk and assign it to a token
+    /// let token : rpc_auth::Token = rpc_auth::Cookie::try_load(&cookie_hint.data_directory).await.unwrap().into();
```

Comment:
I don't see `cookie_hint` defined.   Does this actually compile?   I think you are missing a call like:

```
    let cookie_hint = client.cookie_hint(context::current()).await??;
```

anyway, let's comment out all the setup stuff, so only the RPC call displays in the generated docs.

also, please replace the unwrap() with `?` if possible.